hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -1134,8 +1126,9 @@ protected void validateRollback(
       String commitToRollbackInstantTime,
       HoodieInstant compactionInstant,
       HoodieTimeline deltacommitsSinceCompaction) {
-    // The commit being rolled back should not be earlier than the latest compaction on the MDT. Compaction on MDT only occurs when all actions
-    // are completed on the dataset. Hence, this case implies a rollback of completed commit which should actually be handled using restore.
+    // The commit being rolled back should not be earlier than the latest compaction on the MDT because the latest file slice does not change after all.
+    // Compaction on MDT only occurs when all actions are completed on the dataset.
```

Comment:
> “Compaction on MDT only occurs when all actions are completed on the dataset.”

Can this be relaxed after all DT table service instants use new instant time in MDT and MDT compaction just considers the ones corresponding to the completed instants on DT timeline?

Reply:
In this way, MDT compaction can happen at any time, and any deletions will be reflected in new instant in MDT timeline.

Reply:
That was the first change of this PR, but there is a case for RLI rollback now, the RLI does not include the index mappings in the commit metadata file(instead it is committed along with it), the only way to rollback the RLI mapping is to trigger the `#rollback` action on the `MDT` itself, that is why this restriction is kept.

We can remove it if we can persist the RLI metadata somewhere and figure out the tombstone msgs on the fly while do rollback, but that storage might be costly, I have no clear solution yet, so just keep it as is.